external-builds/pytorch/pytorch_utils.py

```diff
@@ -229,85 +220,60 @@ def get_all_supported_devices(
             result[arch] = []
         result[arch].append(sys_idx)
 
-    if log:
-        print(f"Detected PyTorch supported architecture at device indices: {result}")
+    print(f"Detected PyTorch supported architecture at device indices: {result}")
     return result
 
 
-def get_unique_supported_devices(
-    amdgpu_family: str = "", log: bool = False
-) -> dict[str, int]:
+def get_unique_supported_devices(amdgpu_family: str = "") -> dict[str, list[int]]:
     """
     Returns a dictionary mapping each supported architecture to a single device index (the first one for each).
     This is a convenience wrapper over get_all_supported_devices for situations where
     only one device per arch is desired.
 
     Args:
         amdgpu_family: Optionally filter by a specific AMDGPU family string or pattern.
-        log: If True, passes through to get_all_supported_devices for printing.
 
     Returns:
-        Dictionary: {arch: device_index} for each supported arch.
+        Dictionary: {arch: [device_index]} for each supported arch (single-element lists).
     """
-    devices_by_arch = get_all_supported_devices(amdgpu_family, log=log)
+    devices_by_arch = get_all_supported_devices(amdgpu_family)
     unique_devices = {
-        arch: indices[0] for arch, indices in devices_by_arch.items() if indices
+        arch: [indices[0]] for arch, indices in devices_by_arch.items() if indices
     }
     return unique_devices
 
 
-def get_unique_supported_devices_count(
-    amdgpu_family: str = "", log: bool = False
-) -> int:
-    """Get the number of unique supported architectures.
-
-    Args:
-        amdgpu_family: AMDGPU family filter string (optional).
-        log: If True, passes through to get_unique_supported_devices for printing.
-
-    Returns:
-        Count of unique architectures (one device per arch).
-    """
-    unique_devices_per_arch = get_unique_supported_devices(amdgpu_family, log=log)
-    return len(unique_devices_per_arch)
-
-
 def set_gpu_execution_policy(
-    amdgpu_family: str, policy: str, offset: int = 0, log: bool = True
+    supported_devices: dict[str, list[int]],
+    policy: str,
+    offset: int = 0,
 ) -> list[tuple[str, int]]:
     """
     Configures the HIP_VISIBLE_DEVICES environment variable according to a GPU selection policy,
     enabling targeted execution on specific AMD GPUs for PyTorch/pytest runs. This must be run
     *before* torch is imported, because HIP_VISIBLE_DEVICES cannot affect CUDA device visibility after initialization.
 
     Args:
-        amdgpu_family (str): (Optional) AMDGPU family filter string, e.g., "gfx942", "gfx94X", or "" for all.
+        supported_devices (dict[str, list[int]]): Dictionary mapping architecture names to lists of device indices.
```

Comment:
please have an example how this should look like and architecture names of what?